packages/drivers/routerlicious-driver/src/wholeSummaryDocumentStorageService.ts

```diff
@@ -226,4 +249,77 @@ export class WholeSummaryDocumentStorageService implements IDocumentStorageServi
         });
         await Promise.all(blobCachePutPs);
     }
+
+    // Todo: surface through @fluidframework/server-services-client
+    private convertWholeFlatSummaryToSnapshotTreeAndBlobs(
+        flatSummary: IWholeFlatSummary,
+        stripEntryPath: boolean = true,
+    ): INormalizedWholeSummary {
+        const blobs = new Map<string, ArrayBuffer>();
+        if (flatSummary.blobs) {
+            flatSummary.blobs.forEach((blob) => {
+                blobs.set(blob.id, stringToBuffer(blob.content, blob.encoding ?? "utf-8"));
+            });
+        }
+        const flatSummaryTree = flatSummary.trees && flatSummary.trees[0];
+        const sequenceNumber = flatSummaryTree?.sequenceNumber;
+        const snapshotTree = this.buildHeirarchy(flatSummaryTree, stripEntryPath);
+
+        return {
+            blobs,
+            snapshotTree,
+            sequenceNumber,
+        };
+    }
+
+    // Todo: surface through @fluidframework/server-services-client
```

Comment:
Will remove this after https://github.com/microsoft/FluidFramework/pull/9882 is merged.